localstack/services/apigateway/integration.py

```diff
@@ -0,0 +1,49 @@
+import json
+import logging
+
+from localstack import config
+from localstack.services.apigateway.apigateway_listener import apply_template
+from localstack.services.apigateway.context import ApiInvocationContext
+from localstack.utils.aws import aws_stack
+from localstack.utils.common import make_http_request, to_str
+
+LOG = logging.getLogger(__name__)
+
+
+class BackendIntegration:
+    """
+    Backend integration
+    """
+
+
+class SnsIntegration(BackendIntegration):
+    __slots__ = ["invocation_context"]
+
+    def __init__(self, invocation_context: ApiInvocationContext):
+        self.invocation_context = invocation_context
+
+    def invoke(self):
+        try:
+            data = self.invocation_context.data
+            data = json.dumps(data) if isinstance(data, (dict, list)) else to_str(data)
```

Comment:
nit: Guessing this could become a common pattern - we could also make this a small util function (e.g., `data_as_str(..)`) of the `ApiInvocationContext` class itself 👍

Reply:
Yes, agree! I'm planning more changes on `ApiInvocationContext`.